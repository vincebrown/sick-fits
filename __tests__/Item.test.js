import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import Item from '../components/Item'

const fakeItem = {
  id: 'abcd123',
  title: 'A Cool Item',
  description: 'This item is really cool.',
  image: 'item.jpg',
  largeImage: 'itemLarge.jpg',
  price: 4000,
}

describe('<Item/>', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<Item item={fakeItem} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})

// Very brittle if we change layout of component tests will break, will want to use snapshots.
// describe('<Item/>', () => {
//   const wrapper = shallow(<Item item={fakeItem} />)
//   it('renders the title', () => {
//     const title = wrapper.find('Title a').text()
//     expect(title).toBe(fakeItem.title)
//   })

//   it('renders the description', () => {
//     const description = wrapper.find('p').text()
//     expect(description).toBe(fakeItem.description)
//   })

//   it('renders the image', () => {
//     const img = wrapper.find('img')
//     expect(img.props().src).toBe(fakeItem.image)
//     expect(img.props().alt).toBe(fakeItem.title)
//   })

//   it('renders the price', () => {
//     const PriceTag = wrapper.find('PriceTag')
//     expect(PriceTag.children().text()).toBe('$50')
//   })

//   it('renders the buttons', () => {
//     const buttonList = wrapper.find('.buttonList')
//     expect(buttonList.children()).toHaveLength(3)
//     expect(buttonList.find('Link').exists()).toBe(true)
//     expect(buttonList.find('AddToCart').exists()).toBe(true)
//     expect(buttonList.find('DeleteItem').exists()).toBe(true)
//   })
// })
