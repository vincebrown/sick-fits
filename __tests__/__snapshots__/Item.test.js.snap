// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="A Cool Item"
    src="item.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "abcd123",
          },
        }
      }
    >
      <a>
        A Cool Item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $40
  </PriceTag>
  <p>
    This item is really cool.
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "abcd123",
          },
        }
      }
    >
      <a>
        Edit ✏️
      </a>
    </Link>
    <AddToCart
      id="abcd123"
    />
    <DeleteItem
      id="abcd123"
    >
      Delete This Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
